spec/queries/control_rate_query_spec.rb

```diff
@@ -0,0 +1,117 @@
+  let(:jan_2019) { Time.parse("January 1st, 2019 00:00:00+00:00") }
+  let(:jan_2020) { Time.parse("January 1st, 2020 00:00:00+00:00") }
+  let(:july_2018) { Time.parse("July 1st, 2018 00:00:00+00:00") }
+  let(:july_2020) { Time.parse("July 1st, 2020 00:00:00+00:00") }
+
+  def refresh_views
+    ActiveRecord::Base.transaction do
+      LatestBloodPressuresPerPatientPerMonth.refresh
+      LatestBloodPressuresPerPatientPerQuarter.refresh
+      PatientRegistrationsPerDayPerFacility.refresh
+    end
+  end
+
+  context "monthly period" do
+    it "returns counts for passed in periods" do
+      facility = FactoryBot.create(:facility, facility_group: facility_group_1)
+      patients = [
+        create(:patient, recorded_at: jan_2019, assigned_facility: facility, registration_user: user),
+        create(:patient, status: :dead, recorded_at: jan_2019, assigned_facility: facility, registration_user: user)
+      ]
+
+      Timecop.freeze(june_1_2020) do
+        patients.each do |patient|
+          create(:blood_pressure, :under_control, facility: facility, patient: patient, recorded_at: 2.days.ago, user: user)
+        end
+      end
+
+      refresh_views
+
+      may = query.controlled(facility_group_1, Period.month("May 1 2020")).count
+      june = query.controlled(facility_group_1, june_1_2020.to_period).count
+      august = query.controlled(facility_group_1, Period.month("August 1 2020")).count
+      july = query.controlled(facility_group_1, Period.month("July 1 2020")).count
+
+      expect(may).to eq(2)
+      expect(june).to eq(2)
+      expect(july).to eq(2)
+      expect(august).to eq(0)
+    end
+
+    it "excludes patients who are dead when with_exclusions is true" do
+      facility = FactoryBot.create(:facility, facility_group: facility_group_1)
+      patients = [
+        create(:patient, recorded_at: jan_2019, assigned_facility: facility, registration_user: user),
+        create(:patient, status: :dead, recorded_at: jan_2019, assigned_facility: facility, registration_user: user)
+      ]
+
+      Timecop.freeze(june_1_2020) do
+        patients.each do |patient|
+          create(:blood_pressure, :under_control, facility: facility, patient: patient, recorded_at: 2.days.ago, user: user)
+        end
+      end
+
+      refresh_views
+
+      may = query.controlled(facility_group_1, Period.month("May 1 2020"), with_exclusions: true).count
+      june = query.controlled(facility_group_1, june_1_2020.to_period, with_exclusions: true).count
+      august = query.controlled(facility_group_1, Period.month("August 1 2020"), with_exclusions: true).count
+      july = query.controlled(facility_group_1, Period.month("July 1 2020"), with_exclusions: true).count
+
+      expect(may).to eq(1)
+      expect(june).to eq(1)
+      expect(july).to eq(1)
+      expect(august).to eq(0)
+    end
+  end
+
+  context "quarterly period" do
+    it "returns counts for patients registered in the previous quarter" do
```

Comment:
Does this really mean it returns counts for the _previous_ quarter? It seems like this test shows that querying for Q4 does not return the count for registrations in Q3.